test/dotcom/alerts/affected_stops_test.exs

```diff
@@ -0,0 +1,144 @@
+defmodule Dotcom.Alerts.AffectedStopsTest do
+  use ExUnit.Case
+
+  import Mox
+
+  alias Alerts.{Alert, InformedEntity}
+  alias Dotcom.Alerts.AffectedStops
+  alias Test.Support.{Factories.Stops.Stop, FactoryHelpers}
+
+  describe "affected_stops/1" do
+    test "returns an empty list if there are no informed-entity stops" do
+      # Setup
+      route_id = FactoryHelpers.build(:id)
+
+      Stops.Repo.Mock
+      |> expect(:by_route, fn ^route_id, 0 ->
+        build_random_size_non_empty_stop_list()
+      end)
+
+      alert = Alert.new(informed_entity: [%InformedEntity{stop: nil}])
+
+      # Exercise
+      affected_stops = AffectedStops.affected_stops([alert], [route_id])
+
+      # Verify
+      assert affected_stops == []
+    end
+
+    test "returns stops that are part of the informed entities and part of the route given" do
+      # Setup
+      route_id = FactoryHelpers.build(:id)
+      stops = build_random_size_non_empty_stop_list()
+
+      Stops.Repo.Mock
+      |> stub(:by_route, fn
+        ^route_id, 0 -> build_random_size_stop_list() ++ stops ++ build_random_size_stop_list()
+        _, 0 -> build_random_size_stop_list()
+      end)
+
+      informed_entities = stops |> Enum.map(&%InformedEntity{stop: &1.id})
+      alert = Alert.new(informed_entity: informed_entities)
+
+      # Exercise
+      queried_route_ids = build_random_size_id_list() ++ [route_id] ++ build_random_size_id_list()
+
+      affected_stops = AffectedStops.affected_stops([alert], queried_route_ids)
+
+      # Verify
+      assert affected_stops == stops
+    end
+
+    test "dedupes stops that are part of multiple routes given" do
+      # Setup
+      route_id1 = FactoryHelpers.build(:id)
+      stops1 = build_random_size_non_empty_stop_list()
```

Comment:
suggestion: it might read better if you made one long random stop list first, then extracted various overlapping subsets of that?